packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleTextViewer/SingleTextViewerContainer.js

```diff
@@ -1,121 +1,43 @@
 import PropTypes from 'prop-types'
-import React, { Component } from 'react'
-import request from 'superagent'
+import React from 'react'
+import asyncStates from '@zooniverse/async-states'
 
-import locationValidator from '../../helpers/locationValidator'
 import SingleTextViewer from './SingleTextViewer'
 
-export default class SingleTextViewerContainer extends Component {
-  constructor () {
-    super()
-
-    this.viewer = React.createRef()
-
-    this.state = {
-      content: ''
-    }
-  }
-
-  async componentDidMount () {
-    const { subject } = this.props
-    if (subject) {
-      await this.handleSubject()
-    }
-  }
-
-  async componentDidUpdate (prevProps) {
-    const { subject } = this.props
-    const prevSubjectId = prevProps.subject && prevProps.subject.id
-    const subjectChanged = subject && (subject.id !== prevSubjectId)
-
-    if (subjectChanged) {
-      await this.handleSubject()
-    }
-  }
-
-  getSubjectUrl () {
-    const { subject } = this.props
-    // Find locations that have a text/plain MIME type.
-    const textLocation = subject.locations.find(l => l['text/plain']) || {}
-    const url = Object.values(textLocation)[0]
-    if (url) {
-      return url
-    } else {
-      throw new Error('No text url found for this subject')
-    }
-  }
-
-  async requestData () {
-    const { onError } = this.props
-    try {
-      const url = this.getSubjectUrl()
-      const response = await request.get(url)
-      return response.text || ''
-    } catch (error) {
-      onError(error)
-      return ''
-    }
-  }
-
-  async handleSubject () {
-    const { onError } = this.props
-    try {
-      const rawData = await this.requestData()
-      if (rawData) this.onLoad(rawData)
-    } catch (error) {
-      onError(error)
-      return ''
-    }
-  }
-
-  onLoad (rawData) {
-    const { onReady } = this.props
-    const target = this.viewer.current
-
-    this.setState({
-      content: rawData
-    },
-      function () {
-        onReady({ target })
-      })
-  }
-
-  render () {
-    const {
-      subject
-    } = this.props
-    const { content } = this.state
+const defaultSubject = {
+  content: '',
+  contentLoadingState: asyncStates.initialized
+}
 
-    if (!subject.id) {
-      return null
-    }
+export default function SingleTextViewerContainer ({
+  onError = () => true,
+  onReady = () => true,
+  subject = defaultSubject
+}) {
+  const { content, contentLoadingState } = subject
 
-    if (!content) {
-      return null
-    }
+  if (contentLoadingState === asyncStates.success) {
```

Comment:
Would content ever be an empty string? If not, this can be `if (content)`.

Reply:
This is a great question and sent me down a little rabbit hole. I'd like to punt on this for now, will be opening an issue to document the following and more and will post a link to the issue here asap. 

I think the DigiLeap subjects might include empty text files, but the PFE Project Builder might ignore empty text locations on upload, I need to double check that, and I'm unsure how the CLI handles empty text files, I'll check that too. I'm not sure if that's good or bad, maybe subjects with empty text files should be exclude the text location, maybe they shouldn't and the text location should be the empty text file? Regardless, I'm going to have to dig a little more on this, but for now I think it's ok to keep as is.